source/SIL.AppBuilder.Portal.Frontend/src/ui/routes/admin/settings/stores/edit.tsx

```diff
@@ -1,40 +1,47 @@
-import { withRouter } from 'react-router';
-
-import { query } from '@data';
-
-import { compose, withProps } from 'recompose';
+import React from 'react';
+import { useQuery } from 'react-orbitjs';
 import { paths } from '@ui/routes/admin/paths';
 
 import { useTranslations } from '~/lib/i18n';
 
 import * as toast from '@lib/toast';
 import { useDataActions } from '@data/containers/resources/store/use-data-actions';
 
-import { buildOptions, buildFindRecord, withLoader } from '~/data';
+import { buildOptions, buildFindRecord } from '~/data';
 
 import Form from './-components/form';
 
-export default compose(
-  withRouter,
-  query(({ match: { params: { id } } }) => ({
+import { useRouter } from '~/lib/hooks';
+
+import { RectLoader } from '~/ui/components/loaders';
+
+export default function EditForm() {
+  const { t } = useTranslations();
+  const { update } = useDataActions();
+  const { history, match } = useRouter();
+  const { id } = match.params || {};
+
+  const {
+    result: { store },
+    isLoading,
+  } = useQuery({
     store: [(q) => buildFindRecord(q, 'store', id), buildOptions({ include: ['store-type'] })],
-  })),
-  withLoader(({ store }) => !store),
-  withProps(({ history, store }) => {
-    const { update } = useDataActions();
```

Comment:
the root cause of the issue was that base on whether or not we were in a loading state, we'd conditionally use these hooks.
With hooks, they need to _always_ be used, because of internal implementation details (which could be avoided, imo, but whatever (WeakMap!))